### PR TITLE
Fix bug from introduction of StmtBody

### LINKED ISSUE #685: Fix bug from introduction of StmtBody
The compiler gives an error for this

```
if false {}
```

This is from https://github.com/ballerina-platform/nballerina/commit/cfd71a81e201db489c5c96ba43613029ecd61fb7#r59915488

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose

- Fix code gen for if else stmts 
- Fix stmt block position verifications test not running

Resolves #685

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/front.syntax/ast.bal [hunk 1]
```diff
@@ -107,7 +107,7 @@ public type IfElseStmt record {|
     *PositionFields;
     Expr condition;
     StmtBlock ifTrue;
-    StmtBlock? ifFalse;
+    StmtBlock ifFalse;
```
Comment (jclark) [inline compiler/modules/front.syntax/ast.bal]: Why are you doing this?  The way it was is fine. You just need to fix codeGen.bal.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front.syntax/ast.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/ast.bal]: fixed

File: compiler/modules/front.syntax/parseStmt.bal [hunk 2]
```diff
@@ -339,7 +339,7 @@ function parseIfElseStmt(Tokenizer tok, Position startPos) returns IfElseStmt|er
             IfElseStmt elseIfStmt = check parseIfElseStmt(tok, ifFalseStartPos);
             Position blockStartPos = elseIfStmt.ifTrue.startPos;
             StmtBlock? elseIfFalseBlock = elseIfStmt.ifFalse;
-            Position blockEndPos = (elseIfFalseBlock is StmtBlock)? (elseIfFalseBlock.endPos) : elseIfStmt.ifTrue.startPos;
+            Position blockEndPos = (elseIfFalseBlock is StmtBlock)? (elseIfFalseBlock.endPos) : elseIfStmt.ifTrue.endPos;
```
Comment (jclark) [inline compiler/modules/front.syntax/parseStmt.bal]: Can we simplify this to `(elseIfFalseBlock ?: elseIfStmt.ifTrue).endPos`?

File: compiler/modules/front/codeGen.bal [hunk 3]
```diff
@@ -658,7 +658,8 @@ function codeGenIfElseStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environ
             return codeGenStmts(cx, branchBlock, env, taken);
         }
         else {
-            return cx.semanticErr("if else without a body");
+            // if false whithout else block
+            return { block: branchBlock, assignments: [] };
```
Comment (jclark) [inline compiler/modules/front/codeGen.bal]: You can default the assigments.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front.syntax/parseStmt.bal (hunk 2 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/parseStmt.bal]: fixed

File: compiler/modules/front/codeGen.bal (hunk 3 above)
Comment (heshanpadmasiri) [reply compiler/modules/front/codeGen.bal]: fixed

**Review by jclark — APPROVED** [review]

**Pull request merged**